nautobot/core/ui/object_detail.py

```diff
@@ -378,25 +384,111 @@ def __init__(
+        footer_content_template_path="components/panel/footer_table.html",
         **kwargs,
     ):
         """Instantiate an ObjectsTable panel.
 
         Args:
             table_key (str): The render context key string that contains the BaseTable instance of interest.
+            max_display_count (int, optional):  Maximum number of items to display in the table.
+                If None, defaults to the user's preference or a global setting.
+            include_fields (list, optional): A list of field names to include in the table display.
+                If provided, only these fields will be displayed in the table.
+            exclude_fields (list, optional): A list of field names to exclude from the table display.
+                Cannot be used together with `include_fields`.
+            add_button_route (str, optional): The route used to generate the "add" button URL. Defaults to "default",
+                which uses the default table's model `add` route.
+            add_permissions (list, optional): A list of permissions required for the "add" button to be displayed. If not provided,
+                permissions are determined by default based on the model.
+            related_field_name (str, optional): The name of the field used to filter related objects (typically for query string parameters).
         """
         self.table_key = table_key
+
         super().__init__(
             body_wrapper_template_path=body_wrapper_template_path,
             body_content_template_path=body_content_template_path,
+            header_extra_content_template_path=header_extra_content_template_path,
+            footer_content_template_path=footer_content_template_path,
             **kwargs,
         )
+        self.max_display_count = max_display_count
+        self.add_button_route = add_button_route
+        if exclude_fields and include_fields:
+            raise ValueError("You can only specify either `exclude_fields` or `include_fields`")
+        self.exclude_fields = exclude_fields
+        self.include_fields = include_fields
+        self.related_field_name = related_field_name
+        self.add_permissions = add_permissions
+
+    def _get_table_add_url(self, context):
+        obj = context["obj"]
+        body_content_table_add_url = None
+        request = context["request"]
+        related_field_name = self.related_field_name or obj._meta.model_name
+        return_url = context["return_url"]
+
+        if self.add_button_route == "default":
+            body_content_table = context[self.table_key]
+
+            body_content_table_model = body_content_table.Meta.model
+            permission_name = get_permission_for_model(body_content_table_model, "add")
+            if request.user.has_perms([permission_name]):
+                with contextlib.suppress(NoReverseMatch):
+                    add_route = reverse(get_route_for_model(body_content_table_model, "add"))
```

Comment:
Feels like maybe this should be a try/except that logs a message in the NoReverseMatch case, since it indicates a misconfiguration?